File: 06-arrays-sorting/08_sort_colors.c
#include "../test_runner.h"

void sort_colors(int* nums, int n) {
    /* your code */
}

int main() {
    int a1[] = {2,0,2,1,1,0};
    sort_colors(a1, 6);
    TEST_ARR("basic", a1, ARR(0,0,1,1,2,2), 6);

    int a2[] = {2,0,1};
    sort_colors(a2, 3);
    TEST_ARR("small", a2, ARR(0,1,2), 3);
    TEST_SUMMARY();
}

File: 06-arrays-sorting/08_sort_colors.py
import sys, os; sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
from test_runner import run_tests


# Sort Colors / Dutch National Flag (LC #75) -- Medium
# Sort array of 0s, 1s, 2s in-place, one pass.
#
#   Key: Three pointers: lo (0s boundary), mid (current), hi (2s boundary)

def sort_colors(nums):
    """Sort in-place. Return nums for testing."""
    pass
    return nums


run_tests(sort_colors, [
    (([2,0,2,1,1,0],), [0,0,1,1,2,2]),
    (([2,0,1],),        [0,1,2]),
    (([0],),            [0]),
    (([1,0],),          [0,1]),
])
